core/resource/Repository/ResourceRepository.php

```diff
@@ -89,7 +90,8 @@ public function delete(ContextInterface $context): void
 
         $resourceDeletedEvent = (new ResourceDeleted($resource->getUri()))
             ->setSelectedClass($selectedClass)
-            ->setParentClass($parentClass);
+            ->setParentClass($parentClass)
+            ->setResourceType(reset($resourceType)->getUri());
```

Comment:
please do` reset()` and check `instanceof` it result before calling `->getUri()` because we could get errors here.